pkg/distributions/distributions.go

```diff
@@ -0,0 +1,54 @@
+package distributions
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/llamastack/llama-stack-k8s-operator/pkg/cluster"
+)
+
+// GetDistributionImage returns the image for a given distribution name.
+func GetDistributionImage(distributionName string) (string, error) {
+	clusterInfo := cluster.GetClusterInfo()
+	if clusterInfo == nil {
+		return "", errors.New("cluster info not initialized")
+	}
+
+	image, exists := clusterInfo.DistributionImages[distributionName]
+	if !exists {
+		return "", fmt.Errorf("distribution %s not found in ConfigMap", distributionName)
+	}
+
+	return image, nil
+}
+
+// GetAvailableDistributions returns a map of all available distributions and their images.
+func GetAvailableDistributions() map[string]string {
+	clusterInfo := cluster.GetClusterInfo()
+	if clusterInfo == nil {
+		return nil
+	}
+
+	// Return a copy of the map to prevent external modifications
+	distributions := make(map[string]string, len(clusterInfo.DistributionImages))
```

Comment:
modern go with the standard library `maps` package! It's crazy to me that it took this long to get added to the standard library.

```go
import "maps"

return maps.Clone(clusterInfo.DistributionImages)
```